pkg/jsonmerge/object.go

```diff
@@ -0,0 +1,73 @@
+package jsonmerge
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+
+	"dario.cat/mergo"
+)
+
+type objectMerger struct {
+	io.Writer
+	dst map[string]interface{}
+}
+
+// NewObjectMerger creates a Merger for JSON objects.
+func NewObjectMerger(w io.Writer) Merger {
+	return &objectMerger{
+		Writer: w,
+		dst:    make(map[string]interface{}),
+	}
+}
+
+func (merger *objectMerger) NewPage(r io.Reader, isLastPage bool) io.ReadCloser {
```

Comment:
It's a bit funky that the interface for `Merger` takes `isLastPage` when the object merger never uses it.

Reply:
But the `arrayMerger` does, and it'll be different for each page. Interface implementations in any language won't always use every input. Any work to try-cast to different interfaces to make different calls doesn't seem worth the maintenance cost nor compute time given the limited scope. Feel free to reopen if you disagree.

Reply:
Gah, I'm sorry about this review comment. It was very lazily written. I meant to tie it back to the larger discussion of the consequences of the current API command layer implementation on this PR.

> Interface implementations in any language won't always use every input.

I wouldn't go so far as to say that this is never true but I do think that ignored arguments are generally trying to tell us something about the abstraction. In this case I think it's telling us that the calling code is forcing us to share an interface that would ideally be avoided. Yes, `arrayMerger` uses it because the REST pagination needs it, but it's only required on this interface because the calling code path is cramming graphql and REST together.

I don't think this interface is the problem, I think it's a symptom of a problem elsewhere.

I'm content to leave this resolved though.